src/include/catalog/column_stats_catalog.h

```diff
@@ -107,6 +107,7 @@ class ColumnStatsCatalog : public AbstractCatalog {
 
  private:
   ColumnStatsCatalog(concurrency::TransactionContext *txn);
+  std::vector<oid_t> all_column_ids = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
 

```

Comment:
It seems the purpose of all_column_ids is to use it a copy constructor to construct a vector with all the column oid's. Firstly, I don't think you need to declare an all_column_ids vector, it can be easily constructed explicitly at runtime.  Moreover hardcoding the values it is not a good idea since anyone changing the schema of the catalog table must also remember to update this. 

 A better way to construct this at runtime without breaking abstractions is by using the "num_columns" method and filling the vector using std::iota.